### PR TITLE
Update local setup

### PR DESCRIPTION
**Pull request by claravanstaden**
Snowbridge companion: https://github.com/Snowfork/snowbridge/pull/1131
Resolves: [SNO-826](https://linear.app/snowfork/issue/SNO-826)

- [x] adds snowbridge crates
- [x] removes `parachain` directory
- [x] replaces the Github actions workflows from Parity with our CI, update the CI
- [x] adds script to contribute back upstream (replaces our CI with Parity CI, cleans up code that shouldn't be contributed upstream) - `./bridges/snowbridge/scripts/contribute-upstream.sh <new-branch-name>`

Created https://github.com/paritytech/polkadot-sdk/pull/3186 in the meanwhile to cleanup the directory structure on the `polkadot-sdk`.

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -1,75 +1,2 @@
...
-
-# NPoS and election
-/substrate/frame/election-provider-multi-phase/ @paritytech/staking-core @paritytech/docs-audit
-/substrate/frame/election-provider-support/ @paritytech/staking-core @paritytech/docs-audit
-/substrate/frame/elections-phragmen/ @paritytech/staking-core @paritytech/docs-audit
-/substrate/frame/nomination-pools/ @paritytech/staking-core @paritytech/docs-audit
-/substrate/frame/staking/ @paritytech/staking-core @paritytech/docs-audit
-/substrate/primitives/npos-elections/ @paritytech/staking-core @paritytech/docs-audit
+*.sol @snowfork/snowbridge-solidity-code-owners
+*.rs @snowfork/snowbridge-rust-code-owners
```
Comment (claravanstaden) [inline .github/CODEOWNERS:2]: Just have to add these teams on our repo.

File: rust-toolchain.toml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+[toolchain]
+# rustup has no subcommand (yet) for installing the toolchain in rust-toolchain.toml:
+# https://github.com/rust-lang/rustup/issues/2686
+# The auto-installation behaviour in rustup will likely be removed:
+# https://github.com/rust-lang/rustup/issues/1397
+channel = "1.75.0"
```
Comment (claravanstaden) [inline rust-toolchain.toml:6]: Parity uses the latest stable, so upgraded our toolchain to latest stable. We could also just change this to "stable" in the future.

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -1,75 +1 @@
...
-/substrate/frame/contracts/ @athei @paritytech/docs-audit
-
-# NPoS and election
-/substrate/frame/election-provider-multi-phase/ @paritytech/staking-core @paritytech/docs-audit
-/substrate/frame/election-provider-support/ @paritytech/staking-core @paritytech/docs-audit
-/substrate/frame/elections-phragmen/ @paritytech/staking-core @paritytech/docs-audit
-/substrate/frame/nomination-pools/ @paritytech/staking-core @paritytech/docs-audit
-/substrate/frame/staking/ @paritytech/staking-core @paritytech/docs-audit
-/substrate/primitives/npos-elections/ @paritytech/staking-core @paritytech/docs-audit
+bridges/snowbridge/*.rs @snowfork/snowbridge-devs
```
Comment (vgeddes) [inline .github/CODEOWNERS:1]: Looks like github won't allow external code owners.

Should probably ask Parity if they can add us as limited permission guests to their github org, so they can create a codeowner group.

File: bridges/snowbridge/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,54 @@
+[workspace.package]
+authors = ["Snowfork <contact@snowfork.com>"]
+edition = "2021"
+repository = "https://github.com/snowfork/snowbridge.git"
```
Comment (vgeddes) [inline bridges/snowbridge/Cargo.toml:4]: Needs to be updated

File: .github/CODEOWNERS (hunk 3 above)
Comment (alistair-singh) [inline .github/CODEOWNERS:1]: Does this need to start with `/`?

File: bridges/snowbridge/README.md [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+(https://codecov.io/gh/Snowfork/snowbridge)
+![GitHub](https://img.shields.io/github/license/Snowfork/snowbridge)
+
+Snowbridge is a trustless bridge between Polkadot and Ethereum. For documentation, visit https://docs.snowbridge.network.
+
+## Components
+
+### Parachain
+
+Polkadot parachain and our pallets. See [parachain/README.md](https://github.com/Snowfork/snowbridge/blob/main/parachain/README.md).
```
Comment (alistair-singh) [inline bridges/snowbridge/README.md:12]: Needs to be pointed to the README in this repo.

File: bridges/snowbridge/README.md [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+[`direnv`](https://direnv.net/docs/installation.html) and use the template `.envrc`:
+
+```sh
+cp .envrc.example .envrc
+direnv allow
+```
+
+### Upgrading the Rust toolchain
+
+Sometimes we would like to upgrade rust toolchain. First update `parachain/rust-toolchain.toml` as required and then
```
Comment (alistair-singh) [inline bridges/snowbridge/README.md:87]: Needs to point to the rust-toolchain in the root of this repo now.

**Review by alistair-singh — COMMENTED** [review]
+1

File: .github/CODEOWNERS (hunk 3 above)
Comment (claravanstaden) [reply .github/CODEOWNERS]: For the paritytech/polkadot-sdk, yes. For our repo, we just need to make our snowbridge-devs team public. Do you have access to do that? 

See attached images.

Screenshot 2024-02-06 at 08 26 57
Screenshot 2024-02-06 at 08 27 58

Comment (claravanstaden) [reply .github/CODEOWNERS]: I don't understand? 🤔

Comment (alistair-singh) [reply .github/CODEOWNERS]: Looking at the substrate examples they start with `/`
```
/bridges/snowbridge/*.rs @snowfork/snowbridge-devs
```

Comment (vgeddes) [reply .github/CODEOWNERS]: Ah ok, I made the team public

Comment (claravanstaden) [reply .github/CODEOWNERS]: I see if the path starts with a slash it means from the root of the project, which is more correct. Updated in https://github.com/Snowfork/polkadot-sdk/pull/110/commits/95f4611771c143a216f7ca488a9322a86ac3a6fd.

Comment (claravanstaden) [reply .github/CODEOWNERS]: Thanks, the CODEOWNERS file is now marked as valid. 👍🏻

**Review by vgeddes — APPROVED** [review]

**Review by yrong — APPROVED** [review]
Nice!

**Review by alistair-singh — APPROVED** [review]
+1

**Pull request merged**